megamek/src/megamek/client/ui/preferences/JComboBoxPreference.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2019-2021 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MegaMek.
+ *
+ * MegaMek is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MegaMek is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MegaMek. If not, see <http://www.gnu.org/licenses/>.
+ */
+package megamek.client.ui.preferences;
+
+import javax.swing.*;
+import java.awt.event.ItemEvent;
+import java.awt.event.ItemListener;
+import java.lang.ref.WeakReference;
+
+public class JComboBoxPreference extends PreferenceElement implements ItemListener {
```

Comment:
What's this class for?

Reply:
This class is used to monitor the selected index of a combo box. We use it in the GM Tools Dialog, and it can be nice every once in a while.

Reply:
Sounds good, let's document that.